### DISCUSSION TITLE
Works on localhost but not when deployed

### DISCUSSION THREAD

**Discussion by jahirfiquitiva** [nextauthjs/next-auth#1480] [Help]
**Describe the bug**
Hi, I am currently working on a PWA and implementing `next-auth`. It works pretty fine in `localhost` but it doesn't when I deploy it to Vercel.

When deployed, after login, the site becomes unresponsive and nothing is shown there.

**Steps to reproduce**
You can access [this website: https://dev.synaptiko.io](https://dev.synaptiko.io) and try to login. The app will freeze and eventually becomes unresponsive.

**Expected behavior**
The site should work the same way as it does in desktop

**Screenshots or error logs**

*Localhost site with user logged in*
<img width="1299" alt="Shot 2021-03-08 at 11-29-21@2x" src="https://user-images.githubusercontent.com/10360816/110351317-b5018e00-8002-11eb-9667-74a01b70f90d.png">

*Deployed site with user logged in*
<img width="1397" alt="Shot 2021-03-08 at 11-36-56@2x" src="https://user-images.githubusercontent.com/10360816/110351372-c21e7d00-8002-11eb-8da8-baf57ef7d021.png">

**Additional context**

*`pages/api/auth/[...nextauth].ts`*
```typescript
const options: InitOptions = {
  providers: [
    Providers.LinkedIn({
      clientId: process.env.LINKEDIN_CLIENT_ID || '',
      clientSecret: process.env.LINKEDIN_CLIENT_SECRET || '',
    }),
    Providers.Discord({
      clientId: process.env.DISCORD_CLIENT_ID || '',
      clientSecret: process.env.DISCORD_CLIENT_SECRET || '',
    }),
  ],
  database: process.env.DATABASE_URL,
  debug: true,
};

export default (req: NextApiRequest, res: NextApiResponse): NextApiFunc =>
  NextAuth(req, res, options);
```

*`pages/_app.tsx`*
```typescript
const App: FC<AppProps> = ({ Component, pageProps }) => (
  <ChakraProvider theme={myTheme}>
    <Head>
      <meta
        name={'viewport'}
        content={
          // eslint-disable-next-line max-len
          'width=device-width,initial-scale=1,minimum-scale=1,maximum-scale=1,user-scalable=no,shrink-to-fit=no,viewport-fit=cover,uc-fitscreen=yes'
        }
      />
    </Head>
    <Provider session={pageProps.session}>
      <Component {...pageProps} />
    </Provider>
  </Chakra>
);
```

*custom `useSynaptikoSession`*
```typescript
const useSynaptikoSession = (): [
  Session | null | undefined,
  boolean,
  boolean,
] => {
  const [session, loading] = useSession();
  const [validSession, setValidSession] = useState(false);

  useMemo(() => {
    const isNotValid =
      (!session && !loading) || !Object.keys(session || {}).length;
    setValidSession(!isNotValid);
  }, [session, loading]);

  return [
    loading ? null : validSession ? session : null,
    typeof window !== 'undefined' && loading,
    validSession,
  ];
};

export default useSynaptikoSession;
```

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by jahirfiquitiva** [comment]
`NEXTAUTH_URL` is set to `https://dev.synaptiko.io`

and the providers are setup like this:

<img width="704" alt="Shot 2021-03-08 at 12-15-05@2x" src="https://user-images.githubusercontent.com/10360816/110356197-fe081100-8007-11eb-9386-03e6ca5b969b.png">

*discord*

<img width="367" alt="Shot 2021-03-08 at 12-15-19@2x" src="https://user-images.githubusercontent.com/10360816/110356218-02ccc500-8008-11eb-9dbb-ea22ae5907d0.png">

*linkedin*

**Comment by mattfranciswork0** [comment]
I have the same issue, were you able to solve it?

**Reply by mattfranciswork0** [reply]
Also, how were you able to get the email from linkedin? Linkedin refuses to return the email value for me.

**Comment by SawkaDev** [comment]
Same issue. how did you fix it?

**Comment by Swanand-Wagh** [comment]
whats the fix?

**Comment by LDrex1** [comment]
Since you are deploying on Vercel, try to remove the NEXTAUTH_URL from the envs and see if it works. Unlike other platforms, vercel will discover that particular variable automatically. Since you said the login works, I think that might just be the issue, if that doesn't work. Then check the error written in the logs and let me know.
